client/src/extension.ts

```diff
@@ -2,188 +2,212 @@
+      module: serverModule,
+      options: debugOptions,
+      transport: TransportKind.ipc,
+    },
+    run: { module: serverModule, transport: TransportKind.ipc },
+  };
+
+  // Options to control the language client
+  const clientOptions: LanguageClientOptions = {
+    // Register the server for yml documents
+    documentSelector: [{ scheme: "file", language: "yml" }],
+    synchronize: {
+      // Synchronize the setting section 'yseopml' to the server
+      configurationSection: "yseopml",
+      // Notify the server about file changes to '.clientrc files contain in the workspace
+      fileEvents: workspace.createFileSystemWatcher("**/.clientrc"),
+    },
+  };
+
+  yseopCliPath = vscode.workspace
+    .getConfiguration("yseopml")
+    .get("pathToYseopCli");
+
+  vscode.workspace.onDidChangeConfiguration((event) => {
+    if (event.affectsConfiguration("yseopml.pathToYseopCli")) {
+      yseopCliPath = vscode.workspace
+        .getConfiguration("yseopml")
+        .get("pathToYseopCli");
+    }
+  });
+
+  const batchCmd = vscode.commands.registerCommand("yseopml.batch", () => {
+    ExecYseopCliCommand(yseopCliPath, "batch");
+  });
+
+  const compileCmd = vscode.commands.registerCommand("yseopml.compile", () => {
+    ExecYseopCliCommand(yseopCliPath, "compile");
+  });
+
+  const testCmd = vscode.commands.registerCommand("yseopml.test", () => {
+    ExecYseopCliCommand(yseopCliPath, "test");
+  });
+
+  const cleanCmd = vscode.commands.registerCommand("yseopml.clean", () => {
+    ExecYseopCliCommand(yseopCliPath, "clean");
+  });
+
+  const cleanallCmd = vscode.commands.registerCommand(
+    "yseopml.cleanall",
+    () => {
+      ExecYseopCliCommand(yseopCliPath, "clean", "--all");
+    },
+  );
+
+  const packageCmd = vscode.commands.registerCommand("yseopml.package", () => {
+    ExecYseopCliCommand(yseopCliPath, "package");
+  });
+
+  const infoCmd = vscode.commands.registerCommand("yseopml.info", () => {
+    ExecYseopCliCommand(yseopCliPath, "info");
+  });
+
+  // Create the language client and start the client.
+  const disposable = new LanguageClient(
+    "yseopml",
+    "Yseop Markup Language language server",
+    serverOptions,
+    clientOptions,
+    true,
+  ).start();
+
+  // Push the disposable to the context's subscriptions so that the
+  // client can be deactivated on extension deactivation.
+  // Also register the custom commands.
+  context.subscriptions.push(
+    disposable,
+    batchCmd,
+    compileCmd,
+    testCmd,
+    cleanCmd,
+    cleanallCmd,
+    packageCmd,
+    infoCmd,
+  );
+}
+
+function updateSettings(response: string): void {
+  if (response === "Update Settings") {
+    vscode.commands.executeCommand("workbench.action.openSettings");
+  }
 }
 
 /**
  * Execute a command of Yseop CLI using the folder from which VSCode has been opened as the KB directory.
- * @param yseopCliPath The absolute path to Yseop CLI executable.
+ * @param cliPath The absolute path to Yseop CLI executable.

```

Comment:
Did the linter yell on this name for some reason?

Reply:
In fact `yseopCliPath` was used both as a global variable and an argument.
So yeah, it yelled because of the name shadowing.